README.md

```diff
@@ -24,93 +33,79 @@ spark-submit --master $MASTERURL \
+  --namespace=default --dry-run=client -o yaml | kubectl apply -f -
 ```
 
 ## Build Fink docker image
 
 ### Download Apache Spark
 
-First, you need to choose the Spark version with which you want to run Fink. See above the compatibility matrix. We recommend to use Spark 3.1.3 for the moment.
+First, you need to choose the Spark version with which you want to run Fink, by setting the `SPARK_VERSION` in `$FINKKUB/conf.sh`. See above the compatibility matrix. We recommend to use Spark 3.1.3 for the moment.
 
-Download Spark and untar it in your prefered location:
+The following command will perform:
+1. Download and untat Spark in your prefered location (set `SPARK_INSTALL_DIR` in `$FINKKUB/conf.sh`)
+2. Install Fink additional files inside Spark
 
 ```bash
 # Assuming Scala 2.11
-SPARK_VERSION=3.1.3
-wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.2.tgz .
-tar -xf ./spark-${SPARK_VERSION}-bin-hadoop3.2.tgz
-export SPARK_HOME=${PWD}/spark-${SPARK_VERSION}-bin-hadoop3.2
-```
-
-### Install Fink additional files inside Spark
-
-Clone this repository somewhere on your computer, and register the location:
-
-```bash
-git clone https://github.com/astrolabsoftware/fink-k8s
-export FINKKUB=${PWD}/fink-k8s
-```
-
-Copy Fink files inside your Spark installation:
-
-```bash
-# base Spark image using openjdk:8
-cp ${FINKKUB}/docker/Dockerfile_fink_base ${SPARK_HOME}/kubernetes/dockerfiles/spark
-
-# Fink image
-cp ${FINKKUB}/docker/Dockerfile_fink ${SPARK_HOME}/kubernetes/dockerfiles/spark/bindings/python
-
-# Extra jars for Fink
-cp ${FINKKUB}/jars/*.jar ${SPARK_HOME}/jars/
-
-# Custom launcher
-cp ${FINKKUB}/bin/docker-image-tool-fink.sh ${SPARK_HOME}/bin/
+# Eventually edit some parameters in the Spark section of $FINKKUB/conf.sh
+$FINKKUB/bin/prereq-install.sh
 ```
 
 We have built an image based on `openjdk:11-jre` instead of the official `openjdk:11-jre-slim` which was not suited for our python environment (we heavily use glibc for example which is not in the alpine version). If you know how to easily build the Fink image using `openjdk:11-jre-slim`, contact us! Note that we do not release an image for R (not used), but feel free to contact us if you need it.
 
 ### Build Fink image
 
-To build the image:
+Set your docker account in the `REPO` variable in `$FINKKUB/conf.sh` if you are not using minikube, `TAG` variable contains fink version and spark version used
+
+To build the image run:
 
 ```bash
-cd ${SPARK_HOME}
 
-# remove the `-m` option if you are not using minikube
-# use your docker account in -r if you are not using minikube
-# tag contains fink version and spark version used
-./bin/docker-image-tool-fink.sh -m -r test -t 2.4_3.1.3 -p ./kubernetes/dockerfiles/spark/bindings/python/Dockerfile_fink build
+# Use the `-m` option only if you are using minikube
+$FINKKUB/bin/docker-image-tool-fink.sh -m
```

Comment:
When I execute this command, it just prints the help. It should build the image as well?

```
$FINKKUB/bin/docker-image-tool-fink.sh -m

+ SPARK_ENV_LOADED=
+ SPARK_SCALA_VERSION=
+++ dirname /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh
++ cd /home/peloton/codes/fink-k8s/bin
++ pwd -P
+ readonly DIR=/home/peloton/codes/fink-k8s/bin
+ DIR=/home/peloton/codes/fink-k8s/bin
++ readlink -f /home/peloton/codes/fink-k8s/bin/..
+ readonly FINKKUB=/home/peloton/codes/fink-k8s
+ FINKKUB=/home/peloton/codes/fink-k8s
+ . /home/peloton/codes/fink-k8s/conf.sh
++ K8S_VERSION=v1.25.1
++ CPUS=4
++ readonly SPARK_VERSION=3.1.3
++ SPARK_VERSION=3.1.3
++ readonly SPARK_NAME=spark-3.1.3-bin-hadoop3.2
++ SPARK_NAME=spark-3.1.3-bin-hadoop3.2
++ readonly SPARK_INSTALL_DIR=/home/peloton/fink-k8s-tmp
++ SPARK_INSTALL_DIR=/home/peloton/fink-k8s-tmp
++ export SPARK_HOME=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2
++ SPARK_HOME=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2
++ export PATH=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/bin:/home/peloton/anaconda3/bin:/home/peloton/anaconda3/condabin:/home/peloton/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/peloton/codes/fink-broker/bin:/home/peloton/codes/fink-client/fink_client/scripts:/home/peloton/workspace/spark-3.1.3-bin-hadoop3.2/bin
++ PATH=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/bin:/home/peloton/anaconda3/bin:/home/peloton/anaconda3/condabin:/home/peloton/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/peloton/codes/fink-broker/bin:/home/peloton/codes/fink-client/fink_client/scripts:/home/peloton/workspace/spark-3.1.3-bin-hadoop3.2/bin
++ readonly REPO=local
++ REPO=local
++ readonly TAG=2.4_3.1.3
++ TAG=2.4_3.1.3
++ readonly FINK_IMAGE_NAME=finkk8sdev
++ FINK_IMAGE_NAME=finkk8sdev
++ readonly FINK_K8S_IMAGE=local/finkk8sdev:2.4_3.1.3
++ FINK_K8S_IMAGE=local/finkk8sdev:2.4_3.1.3
++ KAFKA_SOCKET=128.0.0.1:24499
++ KAFKA_TOPIC=changemetopicname
+ . /home/peloton/codes/fink-k8s/bin/include.sh
+ '[' -z /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2 ']'
+ . /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/bin/load-spark-env.sh
++ '[' -z /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2 ']'
++ SPARK_ENV_SH=spark-env.sh
++ '[' -z '' ']'
++ export SPARK_ENV_LOADED=1
++ SPARK_ENV_LOADED=1
++ export SPARK_CONF_DIR=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/conf
++ SPARK_CONF_DIR=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/conf
++ SPARK_ENV_SH=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/conf/spark-env.sh
++ [[ -f /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/conf/spark-env.sh ]]
++ '[' -z '' ']'
++ SCALA_VERSION_1=2.13
++ SCALA_VERSION_2=2.12
++ ASSEMBLY_DIR_1=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/assembly/target/scala-2.13
++ ASSEMBLY_DIR_2=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/assembly/target/scala-2.12
++ ENV_VARIABLE_DOC=https://spark.apache.org/docs/latest/configuration.html#environment-variables
++ [[ -d /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/assembly/target/scala-2.13 ]]
++ [[ -d /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/assembly/target/scala-2.13 ]]
++ export SPARK_SCALA_VERSION=2.12
++ SPARK_SCALA_VERSION=2.12
+ CTX_DIR=/home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/target/tmp/docker
+ trap cleanup_ctx_dir EXIT
+ [[ -m = *--help ]]
+ [[ -m = *-h ]]
+ BASEDOCKERFILE=
+ PYDOCKERFILE=
+ RDOCKERFILE=
+ NOCACHEARG=
+ BUILD_PARAMS=
+ SPARK_UID=
+ CROSS_BUILD=false
+ getopts f:p:R:mXnb:u: option
+ case "${option}" in
+ which minikube
+ minikube status
++ minikube docker-env --shell bash
+ eval export 'DOCKER_TLS_VERIFY="1"' export 'DOCKER_HOST="tcp://192.168.49.2:2376"' export 'DOCKER_CERT_PATH="/home/peloton/.minikube/certs"' export 'MINIKUBE_ACTIVE_DOCKERD="minikube"' '#' To point your shell to 'minikube'\''s' docker-daemon, run: '#' eval '$(minikube' -p minikube 'docker-env)'
++ export DOCKER_TLS_VERIFY=1 export DOCKER_HOST=tcp://192.168.49.2:2376 export DOCKER_CERT_PATH=/home/peloton/.minikube/certs export MINIKUBE_ACTIVE_DOCKERD=minikube
++ DOCKER_TLS_VERIFY=1
++ DOCKER_HOST=tcp://192.168.49.2:2376
++ DOCKER_CERT_PATH=/home/peloton/.minikube/certs
++ MINIKUBE_ACTIVE_DOCKERD=minikube
+ getopts f:p:R:mXnb:u: option
+ case "${@: -1}" in
+ usage
+ cat
Usage: /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh [options] [command]
Builds or pushes the built-in Spark Docker image.

Commands:
  build       Build image. Requires a repository address to be provided if the image will be
              pushed to a different registry.
  push        Push a pre-built image to a registry. Requires a repository address to be provided.

Options:
  -f file               Dockerfile to build for JVM based Jobs. By default builds the Dockerfile shipped with Spark.
  -p file               (Optional) Dockerfile to build for PySpark Jobs. Builds Python dependencies and ships with Spark.
                        Skips building PySpark docker image if not specified.
  -R file               (Optional) Dockerfile to build for SparkR Jobs. Builds R dependencies and ships with Spark.
                        Skips building SparkR docker image if not specified.
  -m                    Use minikube's Docker daemon.
  -n                    Build docker image with --no-cache
  -u uid                UID to use in the USER directive to set the user the main Spark process runs as inside the
                        resulting container
  -X                    Use docker buildx to cross build. Automatically pushes.
                        See https://docs.docker.com/buildx/working-with-buildx/ for steps to setup buildx.
  -b arg                Build arg to build or push the image. For multiple build args, this option needs to
                        be used separately for each build arg.

Using minikube when building images will do so directly into minikube's Docker daemon.
There is no need to push the images into minikube in that case, they'll be automatically
available when running applications inside the minikube cluster.

Check the following documentation for more information on using the minikube Docker daemon:

  https://kubernetes.io/docs/getting-started-guides/minikube/#reusing-the-docker-daemon

Examples:
  - Build image in minikube with tag "testing"
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -m -t testing build

  - Build PySpark docker image
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v2.3.0 -p kubernetes/dockerfiles/spark/bindings/python/Dockerfile build

  - Build and push image with tag "v2.3.0" to docker.io/myrepo
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v2.3.0 build
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v2.3.0 push

  - Build and push JDK11-based image with tag "v3.0.0" to docker.io/myrepo
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v3.0.0 -b java_image_tag=11-jre-slim build
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v3.0.0 push

  - Build and push JDK11-based image for multiple archs to docker.io/myrepo
    /home/peloton/codes/fink-k8s/bin/docker-image-tool-fink.sh -r docker.io/myrepo -t v3.0.0 -X -b java_image_tag=11-jre-slim build
    # Note: buildx, which does cross building, needs to do the push during build
    # So there is no separate push step with -X

+ exit 1
+ cleanup_ctx_dir
+ is_dev_build
+ '[' '!' -f /home/peloton/fink-k8s-tmp/spark-3.1.3-bin-hadoop3.2/RELEASE ']'
```

Reply:
The `build` argument was missing in the doc, in addition, I removed the debug flag